src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -234,6 +234,8 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
             var notificationService = document.Project.Solution.Workspace.Services.GetService<INotificationService>();
             var generatedNameTypeParameterSuffix = GetGeneratedNameTypeParameterSuffix(GetTypeParameters(type, extractableMembers), document.Project.Solution.Workspace);
 
+            var fileBanner = syntaxFactsService.GetFileBanner(document.GetSyntaxRootSynchronously(cancellationToken));
```

Comment:
why it is using GetSyntaxRootSynchronously? either give in root or method should changed to Async

Reply:
I think this should be in "GetUnformattedInterfaceDocument", not here.

Reply:
I would prefer this all be async myself :)

Reply:
is fileBanner here needed?